src/main/java/org/embulk/input/jira/util/JiraUtil.java

```diff
@@ -61,21 +66,22 @@ public static void validateTaskConfig(final PluginTask task)
         if (isNullOrEmpty(uri)) {
             throw new ConfigException("JIRA API endpoint could not be empty");
         }
-        HttpURLConnection connection = null;
-        try {
-            URL u = new URL(uri);
-            connection = (HttpURLConnection) u.openConnection();
-            connection.setRequestMethod("GET");
-            connection.getResponseCode();
+        try (CloseableHttpClient client = HttpClientBuilder.create()
+                                            .setDefaultRequestConfig(RequestConfig.custom()
+                                                                                .setConnectTimeout(HTTP_TIMEOUT)
+                                                                                .setConnectionRequestTimeout(HTTP_TIMEOUT)
+                                                                                .setSocketTimeout(HTTP_TIMEOUT)
+                                                                                .setCookieSpec(CookieSpecs.STANDARD)
+                                                                                .build())
+                                            .build()) {
+            HttpGet request = new HttpGet(task.getUri());
+            try (CloseableHttpResponse response = client.execute(request)) {
+                response.getStatusLine().getStatusCode();
+            }
         }
-        catch (IOException e) {
+        catch (IOException | IllegalArgumentException e) {
```

Comment:
Question: In which case `IllegalArgumentException` will be thrown?

Reply:
IllegalArgumentException for the case user transfer URI like `ftp://example.com`
Instead of throwing IllegalArgumentException I catch it to give more user friendly message